### PR TITLE
Added doco for MessageForwarding feature.

### PR DESCRIPTION
**Pull request by indualagarsamy**
Added code level API for configuring message forwarding in v6.
Added the xml and IProvideConfiguration options for message forwarding for v4 & v5.

### CODE REVIEW

File: Snippets/Snippets_4/Forwarding/ForwardingConfig.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<!--startcode configureForwardingUsingXml-->
+<configuration>
+  <configSections>
+    <section name="UnicastBusConfig"
+             type="NServiceBus.Config.UnicastBusConfig, NServiceBus.Core"/>
+  </configSections>
+  <UnicastBusConfig ForwardReceivedMessagesTo="destinationQueue@machine">
+    <MessageEndpointMappings>
```
Comment (SimonCropp) [inline Snippets/Snippets_4/Forwarding/ForwardingConfig.xml]: is this required?

File: nservicebus/messaging/forwarding.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+Use this feature to forward successfully processed messages from an endpoint to a specified destination endpoint. If there is a need to collect all successfully processed messages from all endpoints, then use the [Auditing Feature](/nservicebus/operations/auditing.md) instead.
+
+### Using Code
+
+snippet:ForwardingWithCode
+
+### Using app.config
+
+WARNING: The ForwardReceivedMessagesTo attribute in the UnicastBusConfig configuration section is no longer supported starting from Version 6. Please switch to the code first API when using Version 6 and above.
```
Comment (SimonCropp) [inline nservicebus/messaging/forwarding.md]: make ForwardReceivedMessagesTo and UnicastBusConfig code

File: nservicebus/messaging/forwarding.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+snippet:configureForwardingUsingXml
+
+### Using IProvideConfiguration
+
+WARNING: The ForwardReceivedMessagesTo attribute in the UnicastBusConfig configuration section is no longer supported starting from Version 6. Please switch to the code first API when using Version 6 and above.
```
Comment (SimonCropp) [inline nservicebus/messaging/forwarding.md]: make code as above

File: Snippets/Snippets_6/Forwarding/Usage.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿
```
Comment (SimonCropp) [inline Snippets/Snippets_6/Forwarding/Usage.cs]: newline

**Comment by SimonCropp** [comment]
u need one of these http://docs.particular.net/nservicebus/errors/automatic-retries#first-level-retries-configuring-flr-through-configurationsource

**Comment by SimonCropp** [comment]
missing menu item

File: nservicebus/messaging/forwarding.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+---
+title: Message forwarding
+summary: Describes how to set up message forwarding
+tags:
+- Forwarding Messages
+redirects:
+---
+
+
+Use this feature to forward successfully processed messages from an endpoint to a specified destination endpoint. If there is a need to collect all successfully processed messages from all endpoints, then use the [Auditing Feature](/nservicebus/operations/auditing.md) instead.
```
Comment (timbussmann) [inline nservicebus/messaging/forwarding.md]: I think this doesn't explain the difference between forwarding and auditing clear enough.

Comment (indualagarsamy) [inline nservicebus/messaging/forwarding.md]: I'm all ears for suggestions. Since this PR has already been merged, create a new PR?

**Pull request merged**